ok, a few more to go...
```
There were 3 failures:

1) OCA\Files_Versions\Tests\VersioningTest::testRename
Failed asserting that false is true.

/srv/www/htdocs/owncloud/apps/files_versions/tests/VersioningTest.php:345

2) OCA\Files_Versions\Tests\VersioningTest::testRenameInSharedFolder
Failed asserting that false is true.

/srv/www/htdocs/owncloud/apps/files_versions/tests/VersioningTest.php:393

3) OCA\Files_Versions\Tests\VersioningTest::testCopy
Failed asserting that false is true.

/srv/www/htdocs/owncloud/apps/files_versions/tests/VersioningTest.php:556

FAILURES!
Tests: 20, Assertions: 153, Failures: 3.
make: *** [Makefile:179: test-php] Error 1
```

But first: 🍔

--- a/apps/files_trashbin/lib/Trashbin.php
+++ b/apps/files_trashbin/lib/Trashbin.php
@@ -678,6 +678,10 @@ class Trashbin {
 			return 0;
 		}
 		$quota = $userObject->getQuota();
+		if ($quota === null || $quota === 'default') {
+			$quota = \OC::$server->getConfig()->getAppValue('files', 'default_quota', null);
+		}
+
 		if ($quota === null || $quota === 'none') {
 			$quota = Filesystem::free_space('/');
 			$softQuota = false;

--- a/apps/files_trashbin/lib/Trashbin.php
+++ b/apps/files_trashbin/lib/Trashbin.php
@@ -677,12 +677,8 @@ class Trashbin {
 		if(is_null($userObject)) {
 			return 0;
 		}
-		$quota = $userObject->getQuota();
-		if ($quota === null || $quota === 'default') {
-			$quota = \OC::$server->getConfig()->getAppValue('files', 'default_quota', null);
-		}
-
-		if ($quota === null || $quota === 'none') {
+		$quota = \OC_Util::getUserQuota($userObject);
+		if ($quota === \OCP\Files\FileInfo::SPACE_UNLIMITED) {
 			$quota = Filesystem::free_space('/');
 			$softQuota = false;
 			// inf or unknown free space

--- a/apps/files_versions/lib/Storage.php
+++ b/apps/files_versions/lib/Storage.php
@@ -715,11 +715,8 @@ class Storage {
 			$versionsFileview = new View('/'.$uid.'/files_versions');
 
 			$softQuota = true;
-			$quota = $user->getQuota();
-			if ($quota === null || $quota === 'default') {
-				$quota = $config->getAppValue('files', 'default_quota', null);
-			}
-			if ($quota === null || $quota === 'none') {
+			$quota = \OC_Util::getUserQuota($user);
+			if ($quota === \OCP\Files\FileInfo::SPACE_UNLIMITED) {
 				$quota = Filesystem::free_space('/');
 				$softQuota = false;
 			} else {

--- a/apps/files_versions/lib/Storage.php
+++ b/apps/files_versions/lib/Storage.php
@@ -716,7 +716,10 @@ class Storage {
 
 			$softQuota = true;
 			$quota = $user->getQuota();
-			if ( $quota === null || $quota === 'none' ) {
+			if ($quota === null || $quota === 'default') {
+				$quota = $config->getAppValue('files', 'default_quota', null);
+			}
+			if ($quota === null || $quota === 'none') {
 				$quota = Filesystem::free_space('/');
 				$softQuota = false;
 			} else {

--- a/lib/private/User/User.php
+++ b/lib/private/User/User.php
@@ -362,7 +362,7 @@ class User implements IUser {
 	public function getQuota() {
 		$quota = $this->account->getQuota();
 		if(is_null($quota)) {
-			$quota = $this->config->getAppValue('files', 'default_quota', 'default');
+			return 'default';
 		}
 		return $quota;
 	}

--- a/lib/private/User/User.php
+++ b/lib/private/User/User.php
@@ -362,7 +362,7 @@ class User implements IUser {
 	public function getQuota() {
 		$quota = $this->account->getQuota();
 		if(is_null($quota)) {
-			$quota = $this->config->getAppValue('files', 'default_quota', 'none');
+			$quota = $this->config->getAppValue('files', 'default_quota', 'default');
 		}
 		return $quota;
 	}

--- a/lib/private/legacy/util.php
+++ b/lib/private/legacy/util.php
@@ -361,11 +361,15 @@ class OC_Util {
 	/**
 	 * Get the quota of a user
 	 *
-	 * @param string $userId
+	 * @param string|IUser $userId
 	 * @return int Quota bytes
 	 */
 	public static function getUserQuota($userId) {
-		$user = \OC::$server->getUserManager()->get($userId);
+		if ($userId instanceof IUser) {
+			$user = $userId;
+		} else {
+			$user = \OC::$server->getUserManager()->get($userId);
+		}
 		if (is_null($user)) {
 			return \OCP\Files\FileInfo::SPACE_UNLIMITED;
 		}

--- a/lib/private/legacy/util.php
+++ b/lib/private/legacy/util.php
@@ -370,7 +370,10 @@ class OC_Util {
 			return \OCP\Files\FileInfo::SPACE_UNLIMITED;
 		}
 		$userQuota = $user->getQuota();
-		if($userQuota === 'none') {
+		if ($userQuota === null || $userQuota === 'default') {
+			$userQuota = \OC::$server->getConfig()->getAppValue('files', 'default_quota', 'none');
+		}
+		if ($userQuota === null || $userQuota === 'none') {
 			return \OCP\Files\FileInfo::SPACE_UNLIMITED;
 		}
 		return OC_Helper::computerFileSize($userQuota);

